addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/handler/SatelBridgeHandler.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.satel.handler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.ConfigStatusBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.satel.internal.command.NewStatesCommand;
+import org.openhab.binding.satel.internal.command.SatelCommand;
+import org.openhab.binding.satel.internal.config.SatelBridgeConfig;
+import org.openhab.binding.satel.internal.event.ConnectionStatusEvent;
+import org.openhab.binding.satel.internal.event.SatelEvent;
+import org.openhab.binding.satel.internal.event.SatelEventListener;
+import org.openhab.binding.satel.internal.protocol.SatelModule;
+import org.openhab.binding.satel.internal.types.IntegraType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SatelBridgeHandler} is base class for all bridge handlers.
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ */
+public abstract class SatelBridgeHandler extends ConfigStatusBridgeHandler implements SatelEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(SatelBridgeHandler.class);
+
+    private SatelBridgeConfig config;
+    private SatelModule satelModule;
+    private ScheduledFuture<?> pollingJob;
+    private String userCodeOverride;
+
+    public SatelBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void incomingEvent(SatelEvent event) {
+        if (event instanceof ConnectionStatusEvent) {
+            ConnectionStatusEvent statusEvent = (ConnectionStatusEvent) event;
+            // update bridge status and get new states from the system
+            if (statusEvent.isConnected()) {
+                updateStatus(ThingStatus.ONLINE);
+                satelModule.sendCommand(new NewStatesCommand(satelModule.getIntegraType().hasExtPayload()));
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        statusEvent.getReason());
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // the bridge does not support any command at the moment
+        logger.debug("New command for {}: {}", channelUID, command.toFullString());
+    }
+
+    protected void initialize(final SatelModule satelModule) {
```

Comment:
Where do you ever set the Thing status to ONLINE or OFFLINE here (depending on whether the connection can be established or not)?

Reply:
This is an abstract class, thing status is set in its subclasses.

Reply:
I didn't find it in any of the sub-classes either, that's why I asked.

Reply:
Ethm1BridgeHandler.java line 56, IntRSBridgeHandler.java line 55, SatelBridgeHandler.java lines 56, 59

Reply:
All those places only set it to OFFLINE if some configuration is missing.
My question was: "Where do you ever set the Thing status to ONLINE or OFFLINE here (depending on whether the connection can be established or not)?"

Reply:
Apparently I don't understand something. For me it is "SatelBridgeHandler.java lines 56, 59", but maybe you're asking about something else?

Reply:
Maybe I do not fully grasp the internals of the binding then. Lines 56-59 seem to be only executed if some "event" comes along. My (maybe too simple) assumption was that those are events that happen if some activity is already in place. If you can confirm that right after `initialize()` has finished, there will always be some event (where is it triggered?), which makes lines 56-59 being executed, then all should be fine indeed.

Reply:
That's right. There's an event with current connection status. Once the connection is established and the module sends a response the new "connected" status is broadcasted. The same happens when connection is broken.
This is all done when `SatelModule.open` is called (SatelBridgeHandler, line 77). It may seem a bit complicated, because it is writen as a library with some level of abstraction. For example `SatelBridgeHandler` class is an adapter for `SatelModule`.

Reply:
Ok, then it was my mistake to not see this, sorry.
Please let me know, once you have addressed the other comments and we shouldn't be far away from the merge anymore.